### PR TITLE
[WIP] Split cs.js into modules, clean up content scripts, avoid polluting scope

### PR DESCRIPTION
**Pull request by WorldLanguages**
Because of #2599 we can now use dynamic import on content scripts.
However, dynamic import will slow down how fast code runs, so we need to be careful and only use it for code that isn't time sensitive.

Goals of this PR:
- Shrink the size of cs.js by moving non critical parts to other files. For explanation on why it was so big, see below; we intentionally made it a single file because separate content scripts would share variable scoping even if we use let/const, and not even ESLint would be able to catch potential errors there.
- Avoid content scripts other than cs.js from unintentionally polluting the global extension scope. All content scripts share the same scope unless they're dynamically imported modules, so `if(){}` blocks and IIFEs fix this. ESLint isn't smart enough to know this, so the easiest way to deal with this is to simply make sure only cs.js registers variables and functions at top level.
- Avoid code duplication

This PR will involve some copy pasting of code around, so I'll make sure to clearly mark code that has been changed after copy-pasting with `// comments like this //`

Because of Chrome, we need to add modules that we want to dynamically import on `web_accessible_resources`, even tho they run on extension context. Firefox does not have this limitation.

### CODE REVIEW

File: content-scripts/inject/module.js [hunk 1]
```diff
@@ -12,7 +12,7 @@ scratchAddons.eventTargets = {
 scratchAddons.session = {};
 const consoleOutput = (logAuthor = "[page]") => {
   const style = {
-    // Remember to change these as well on cs.js
+    // Remember to change these as well on cs-console.js
```
Comment (RedGuy12) [inline content-scripts/inject/module.js:15]: cant we make this an external lib already

Comment (WorldLanguages) [reply content-scripts/inject/module.js:15]: Good point, we now could

**Comment by FunctionalMetatable** [comment]
tbh one day it might be a good idea to bundle scripts all into one using rollup/webpack

**Comment by WorldLanguages** [comment]
> tbh one day it might be a good idea to bundle scripts all into one content scripts using rollup/webpack

why?

**Comment by FunctionalMetatable** [comment]
> > tbh one day it might be a good idea to bundle scripts all into one content scripts using rollup/webpack
> 
> why?

we can load addons faster

**Comment by WorldLanguages** [comment]
> we can load addons faster

It won't make a noticeable difference.

**Comment by FunctionalMetatable** [comment]
> > we can load addons faster
> 
> It won't make a noticeable difference.

but its better for addons like scratchblocks and stuff

**Comment by WorldLanguages** [comment]
Dynamic import support for content scripts was added to Firefox since version 89, and we decided to support Firefox 86+ so this can't be done.
Also, there can be speed issues and flashes if themes like dark mode and scratch2.0→3.0 aren't added fast enough, and using dynamic import is probably slower, so I don't think we should use dynamic import() for the text-color library.

**Pull request closed without merging**